northstar/src/runtime/mod.rs

```diff
@@ -206,7 +206,18 @@ async fn runtime_task(config: &'_ Config, stop: CancellationToken) -> Result<(),
             // A container process existed. Check `process::wait_exit` for details.
             Event::Exit(container, exit_status) => state.on_exit(&container, &exit_status).await,
             // The runtime os commanded to shut down and exit.
-            Event::Shutdown => break state.shutdown().await,
+            Event::Shutdown => {
+                log::debug!("Stopping to process events");
+                drop(event_rx);
+
+                log::debug!("Stopping console");
+                if let Some(console) = console.take() {
+                    console.shutdown().await.map_err(Error::Console)?;
+                }
```

Comment:
Shouldn't this be the other way round. Stop the console and it's connections *before* the main loop?

Reply:
The event loop has to be stopped first. This _signals_ the open connections that they won't be getting their responses. Then the next step is to wait for those connections to terminate and close the console. Finally the state is shut down.

Reply:
Conceptually the event loop should remain processing until no further events are generated. 
We need another way to signal open connections that will not be serviced anymore.

Reply:
There is this `shutdown` method on the connections, I will try to invert the order with that.

Reply:
@marcmo, what should be the behavior with a shutdown request followed by n other requests? the remaining request should be processed before the shutdown?